azure-quantum/azure/quantum/projectq/backends/ionq.py

```diff
@@ -0,0 +1,163 @@
+##
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+##
+import json
+import uuid
+from typing import TYPE_CHECKING
+
+from azure.quantum import __version__
+from azure.quantum.projectq.job import (
+    AzureQuantumJob, 
+    IONQ_PROVIDER, 
+    IONQ_INPUT_DATA_FORMAT, 
+    IONQ_OUTPUT_DATA_FORMAT
+)
+
+try:
+    from projectq.backends import IonQBackend as ProjectQIonQBackend
+    import projectq.setups.ionq
+except ImportError:
+    raise ImportError(
+    "Missing optional 'projectq' dependencies. \
+To install run: pip install azure-quantum[projectq]"
+)
+
+if TYPE_CHECKING:
+    from azure.quantum.projectq import AzureQuantumEngine
+
+import logging
+logger = logging.getLogger(__name__)
+
+__all__ = ["IonQBackend", "IonQQPUBackend", "IonQSimulatorBackend"]
+
+
+class IonQBackend(ProjectQIonQBackend):
+    backend_name = None
+
+    def __init__(
+        self, 
+        use_hardware=False, 
+        num_runs=100, 
+        verbose=False, 
+        device="ionq_simulator", 
+        retrieve_execution=None
+    ):
+        """Base class for interfacing with an IonQ backend in Azure Quantum"""
+        logger.info("Initializing IonQBackend")
+
+        super().__init__(
+            use_hardware=use_hardware, 
+            num_runs=num_runs, 
+            verbose=verbose, 
+            device=device, 
+            num_retries=0,
+            interval=1, 
+            retrieve_execution=retrieve_execution
+        )
+        
+    def get_engine_list(self):
+        """Return the default list of compiler engine for the IonQ platform."""
+        return projectq.setups.ionq.get_engine_list(
+            device=self.backend_name
+        )
+
+    def run(self, name=None, **kwargs):
+        """Submits the given circuit to run on an IonQ target."""
+        logger.info(f"Submitting new job for backend {self.device}")
+
+        if name is None:
+            random_suffix = str(uuid.uuid4())[:8]
+            name = "projectq-ionq-circuit-{}".format(random_suffix)
+
+        qubit_mapping = self.main_engine.mapper.current_mapping
+        measured_ids = self._measured_ids[:]
+
+        num_qubits = len(qubit_mapping.keys())
+        meas_map = [qubit_mapping[qubit_id] for qubit_id in measured_ids]
+
+        ionq_circ = self._circuit
+
+        input_data = json.dumps({
+            "qubits": num_qubits,
+            "circuit": ionq_circ,
+        })
+
+        input_params = {
+            "shots": self._num_runs
+        }
+
+        metadata = {
+            "name": name, 
+            "num_qubits": num_qubits, 
+            "meas_map": meas_map,
+            # "num_retries": self._num_retries,
+            # "interval": self._interval
```

Comment:
`num_retries` and `interval` are ProjectQ properties used for retry mechanism in case job failed. This is **client side** implementation in ProjectQ (like a for-loop with sleep).

Since, QDK has it own job management, I choose to drop these arguments from IonQ and Honeywell wrapper class, but I open for any suggestions, let me know your thoughts.

Reply:
I suggest adding these back in and using them instead as timeout for the API call.